framework/aster-frame/src/vm/dma/dma_stream.rs

```diff
@@ -222,6 +222,73 @@ impl HasPaddr for DmaStream {
     }
 }
 
+/// A slice of streaming DMA mapping.
+#[derive(Debug, Clone)]
+pub struct DmaStreamSlice {
+    stream: DmaStream,
```

Comment:
`DmaStreamSlice` is so similar to `DmaStream` that adding the concept of `DmaStreamSlice` only provides limited marginal value. Furthermore, the standard "slice" in Rust means borrowed data, not owned data. This is in conflict with the semantic of `DmaStreamSlice`, which is stilled owned.

For the reasons above, I propose re-defining `DmaStreamSlice` as below.

```rust
pub struct DmaStreamSlice<'a> {
    stream: &'a DmaStream,
    offset: usize,
    len: usize,
}
```

Reply:
Done